agent-control/src/sub_agent/on_host/supervisor.rs

```diff
@@ -60,18 +60,27 @@ impl SupervisorStarter for NotStartedSupervisorOnHost {
     ) -> Result<Self::SupervisorStopper, SupervisorStarterError> {
         let ctx = self.ctx.clone();
 
-        let thread_contexts = vec![
+        let executable_thread_contexts =
+            self.executables.clone().into_iter().map(|e| {
+                self.start_process_thread(sub_agent_internal_publisher.clone(), e.clone())
+            });
+
+        let thread_contexts: Vec<StartedThreadContext> = vec![
             self.start_health_check(sub_agent_internal_publisher.clone())?,
             self.start_version_checker(sub_agent_internal_publisher.clone()),
-            // the process thread is created if exec is Some
-            self.maybe_exec
-                .clone()
-                .map(|e| self.start_process_thread(sub_agent_internal_publisher, e)),
-        ];
+        ]
+        .into_iter()
+        .flatten()
+        .collect();
+
+        let thread_contexts = executable_thread_contexts
+            .into_iter()
+            .chain(thread_contexts)
+            .collect();
 
         Ok(StartedSupervisorOnHost {
             ctx,
-            thread_contexts: thread_contexts.into_iter().flatten().collect(),
+            thread_contexts,
```

Comment:
This is surely for another PR because it could be added earlier, and only if required, but probably using [scoped threads](https://doc.rust-lang.org/std/thread/fn.scope.html) could ease resource sharing and have the guarantee that all are joined before continuing.

Reply:
Do you want to create a task to review?

Reply:
I think that unless its something that we know we will need it's better for an Innovation Day or something. I'll note it down!